lib/pbench/server/api/resources/query_apis/__init__.py

```diff
@@ -296,23 +351,26 @@ def __str__(self) -> str:
 
 class ElasticBase(Resource):
     """
-    ElasticBase A base class for Elasticsearch queries that allows subclasses
-    to provide customers pre- and post- processing.
+    A base class for Elasticsearch queries that allows subclasses to provide
+    custom pre- and post- processing.
 
     This class extends the Flask Resource class in order to connect the post
     and get methods to Flask's URI routing algorithms. It implements a common
     JSON client payload intake and validation, along with the mechanism for
     calling Elasticsearch and processing errors.
 
-    Hooks are defined for subclasses extending this class to "assemble" the
-    Elasticsearch request payload from Pbench server data and the client's
-    JSON payload, and to "postprocess" a successful response payload from
-    Elasticsearch.
+    Hooks are defined for subclasses extending this class to "preprocess"
```

Comment:
"Abstract methods are defined"??? But, yeah, "hooks" probably wasn't the best attempt here.

Reply:
"The `p-`, `a-`, and `p-` methods are provided as hooks to allow subclasses to provide custom processing at key points."  (I think that's a fancy way of saying "this class was designed to be extended by subclasses", which is probably just a fancy way of saying "this is an abstract class"!)  😁

Reply:
Yes, if only Python had the useful concept of explicit abstraction. I'd rather have a compile-time failure when a subclass fails to override an abstract method rather than a runtime `NotYetImplemented` exception; but that's life in the Python zone.

Reply:
Python _does_ have the ability to declare "[abstract base classes](https://docs.python.org/3/library/abc.html)".  I don't remember what the implications of them are (I think it's just that attempting to instantiate one results in an error), but I think they might be worth using in a case like this one.  (The docs also mention "metaclasses", whatever _those_ are....)